### ISSUE TITLE
Switching between profiles is unreliable

### ISSUE THREAD

**Issue by h2dden** [GrapheneOS/os_issue_tracker#435]
I just started to discover the profiles in graphene os and while playing around all the apps and data still reside in the owner profile.
But, when switching from a profile back to the owner profile I experienced some issues which I find difficult to reproduce: these issue are quite sporadic.
Several times, when ending a session and returning to the owner profile the main screen of the owner profile is shown: I have a widget displaying my calendar events, and superimposed is shown a transparent lock screen: basically a lock and a message asking to swip up to unlock. The main point is that while the screen should be locked I can see my calendar in the background. I will keep playing around and when I can reproduce I will post here a picture of the screen.
Another time, when ending a session and returning to the owner profile I could not unlock the screen with the fingerprint: it took me a while to understand that somehow the fingerprints were gone: I had to re-register the fingerprints all over again. This will be difficult to capture in a pic or video

**Comment by thestinger** [comment]
This is an issue with the end session feature rather than switching between profiles. You can switch without ending the session. The end session feature is a hidden Android feature that we make usable by regular users rather than only device management apps. We've reported this problem with the race conditions when ending the session upstream.

**Comment by thestinger** [comment]
I made this specifically about the display issues. There may be other issues, but the display issues are what I reported upstream a month or so ago.

**Comment by h2dden** [comment]
Clear. If needed I can play around until I get the calendar in the background of the locked screen.
But the issue with fingerprints being wiped... I have no idea how to capture. So far it happened only once, while the screen issue three times

**Comment by thestinger** [comment]
You should probably file that as a separate issue. I heard someone mention something like that with Face Unlock before but they were never able to replicate it. I don't think we have an issue for it. I don't really think it's related. Maybe, possibly related, but it seems like a distinct issue even if it is.

**Comment by h2dden** [comment]
as long as you know of the issue, I think is enough
I can see that the current ticket is tagged "upstream"... so I think this is not something you will tackle very soon

**Comment by thestinger** [comment]
Well I reported it upstream as a minor security vulnerability and they acknowledged it.

**Comment by thestinger** [comment]
Please check if this is still an issue.

**Comment by h2dden** [comment]
has not happened in a while so... as it is sporadic and not something I can trigger right now: better close this ticket.
If I come across the bug again, I will open another ticket
